src/include/detail/linalg/tdb_matrix.h

```diff
@@ -147,10 +185,39 @@ class tdbBlockedMatrix : public MatrixBase {
   tdbBlockedMatrix(
       const tiledb::Context& ctx,
       const std::string& uri,
+      size_t first_col,
```

Comment:
Per the extensive comment above, `first_col` et al should be the same type as the domain -- we should probably have a global type alias for this.  In `tdbMatrix` we use `row_domain_type` and `col_domain_type` (both of which are `int32_t` -- but we should not hard code that everywhere).  We could just use those:
```
tdbBlockedMatrix(
       const tiledb::Context& ctx,
       const std::string& uri,
       std::optional<col_domain_type> first_row, // etc
```

Reply:
Sure, created a new story for this as: https://app.shortcut.com/tiledb-inc/story/42696